### PR TITLE
[MySQL] Add Advanced Threat Protection for Azure MySQL

### PR DESCRIPTION
**Pull request by susantabiswas**
**Related command**

Enable / Disable MySQL flexible server's Advanced Threat Protection using the `--state` argument
>az mysql flexible-server advanced-threat-protection-setting update -g <resource_group_name> -n <server_name> --state Disabled

Get MySQL flexible server's Advanced Threat Protection setting
>az mysql flexible-server advanced-threat-protection-setting show -g <resource_group_name> -n <server_name>

**Description**

Added support for Advanced Server Threat Protection Setting for Azure MySQL flexible server. It supports commands to update and show the advanced threat protection setting.

**Testing Guide**

- Automated tests added to cover the new commands

Enable / Disable MySQL flexible server's Advanced Threat Protection using the `--state` argument
>az mysql flexible-server advanced-threat-protection-setting update -g <resource_group_name> -n <server_name> --state Disabled

Get MySQL flexible server's Advanced Threat Protection setting
>az mysql flexible-server advanced-threat-protection-setting show -g <resource_group_name> -n <server_name>

**History Notes**

[MySQL] `az mysql flexible-server advanced-threat-protection-setting show`: Show server's advanced threat protection setting
[MySQL] `az mysql flexible-server advanced-threat-protection-setting update`: Update server's advanced threat protection setting using `--state` as Enabled/Disabled

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Do not Merge

**Review by honghr — APPROVED** [review]

**Comment by yonzhan** [comment]
Please fix CI issues

**Comment by susantabiswas** [comment]
PR description related error fixed

File: src/azure-cli/azure/cli/command_modules/mysql/_help.py [hunk 1]
```diff
@@ -12,6 +12,33 @@
...
 """
 
+helps['mysql flexible-server advanced-threat-protection-setting'] = """
+type: group
+short-summary: Manage the server's advanced threat protection setting.
+"""
+
+helps['mysql flexible-server advanced-threat-protection-setting show'] = """
+type: command
+short-summary: Gets the server's advanced threat protection setting.
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/mysql/_help.py:22]:
```suggestion
short-summary: Get the server's advanced threat protection setting.
```

File: src/azure-cli/azure/cli/command_modules/mysql/_help.py [hunk 2]
```diff
@@ -12,6 +12,33 @@
...
+examples:
+  - name: Get the advanced threat protection setting.
+    text: az mysql flexible-server advanced-threat-protection-setting show -g mygroup -n myserver
+"""
+
+helps['mysql flexible-server advanced-threat-protection-setting update'] = """
+type: command
+short-summary: Updates the server's advanced threat protection setting.
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/mysql/_help.py:30]:
```suggestion
short-summary: Update the server's advanced threat protection setting.
```

**Review by evelyn-ys — APPROVED** [review]

**Review by necusjz — APPROVED** [review]

**Pull request merged**